presto-hive/src/main/java/com/facebook/presto/hive/util/FilterPushdownUtils.java

```diff
@@ -0,0 +1,532 @@
+            ConnectorMetadata metadata,
+            ConnectorSession session)
+    {
+        checkArgument(!FALSE_CONSTANT.equals(filter), "Cannot pushdown filter that is always false");
+
+        ConnectorPushdownFilterResult connectorPushdownFilterResult;
+        if (TRUE_CONSTANT.equals(filter) && currentLayoutHandle.isPresent()) {
+            connectorPushdownFilterResult = new ConnectorPushdownFilterResult(metadata.getTableLayout(session, currentLayoutHandle.get()), TRUE_CONSTANT);
+            return new Result(connectorPushdownFilterResult);
+        }
+
+        DomainTranslator.ExtractionResult<Subfield> decomposedFilter = getDecomposedFilter(rowExpressionService, functionResolution, session, filter, currentLayoutHandle);
+
+        if (decomposedFilter.getTupleDomain().isNone()) {
+            connectorPushdownFilterResult = new ConnectorPushdownFilterResult(EMPTY_TABLE_LAYOUT, FALSE_CONSTANT);
+            return new Result(connectorPushdownFilterResult);
+        }
+
+        RowExpression optimizedRemainingExpression = rowExpressionService.getExpressionOptimizer().optimize(decomposedFilter.getRemainingExpression(), OPTIMIZED, session);
+        if (optimizedRemainingExpression instanceof ConstantExpression) {
+            ConstantExpression constantExpression = (ConstantExpression) optimizedRemainingExpression;
+            if (FALSE_CONSTANT.equals(constantExpression) || constantExpression.getValue() == null) {
+                connectorPushdownFilterResult = new ConnectorPushdownFilterResult(EMPTY_TABLE_LAYOUT, FALSE_CONSTANT);
+                return new Result(connectorPushdownFilterResult);
+            }
+        }
+
+        return new Result(decomposedFilter, optimizedRemainingExpression);
+    }
+
+    public static TupleDomain<ColumnHandle> getEntireColumnDomain(
+            Result result,
+            Map<String, ColumnHandle> columnHandles)
+    {
+        TupleDomain<ColumnHandle> entireColumnDomain = result.getDecomposedFilter().getTupleDomain()
+                .transform(subfield -> isEntireColumn(subfield) ? subfield.getRootName() : null)
+                .transform(columnHandles::get);
+
+        return entireColumnDomain;
+    }
+
+    public static Constraint<ColumnHandle> extractDeterministicConjuncts(
+            RowExpressionService rowExpressionService,
+            StandardFunctionResolution functionResolution,
+            FunctionMetadataManager functionMetadataManager,
+            ConnectorSession session,
+            Result result,
+            Map<String, ColumnHandle> columnHandles,
+            TupleDomain<ColumnHandle> entireColumnDomain,
+            Constraint<ColumnHandle> constraint)
+    {
+        if (!TRUE_CONSTANT.equals(result.getDecomposedFilter().getRemainingExpression())) {
+            LogicalRowExpressions logicalRowExpressions = new LogicalRowExpressions(
+                    rowExpressionService.getDeterminismEvaluator(), functionResolution, functionMetadataManager);
```

Comment:
Presto doesn't use this style. It puts one parameter on each line if the line is too long. Ditto